kitty.go

```diff
@@ -7,31 +7,82 @@ import (
 	"github.com/charmbracelet/x/ansi"
 )
 
+// setKittyKeyboardFlagsMsg is a message to set Kitty keyboard progressive
+// enhancement protocol flags.
+type setKittyKeyboardFlagsMsg int
+
+// EnableKittyKeyboard is a command to enable Kitty keyboard progressive
+// enhancement protocol.
+//
+// The flags parameter is a bitmask of the following
+//
+//	0:  Disable all features
+//	1:  Disambiguate escape codes
+//	2:  Report event types
+//	4:  Report alternate keys
+//	8:  Report all keys as escape codes
+//	16: Report associated text
+//
+// See https://sw.kovidgoyal.net/kitty/keyboard-protocol/ for more information.
+func EnableKittyKeyboard(flags int) Cmd {
+	return func() Msg {
+		return setKittyKeyboardFlagsMsg(flags)
+	}
+}
+
+// DisableKittyKeyboard is a command to disable Kitty keyboard progressive
+// enhancement protocol.
+func DisableKittyKeyboard() Msg {
+	return setKittyKeyboardFlagsMsg(0)
+}
+
+// requestKittyKeyboardFlagsMsg is a message to request the current Kitty keyboard
+// progressive enhancement protocol flags.
+type requestKittyKeyboardFlagsMsg struct{}
+
+// RequestKittyKeyboardFlags is a command to request the current Kitty keyboard
+// progressive enhancement protocol flags from the terminal.
+func RequestKittyKeyboardFlags() Msg {
```

Comment:
Should this be to make it align more with current bubbletea commands?

```suggestion
func KittyKeyboardFlags() Msg {
```

CC/ @caarlos0 @meowgorithm @bashbunni

Reply:
not sure... I think prefixing with `Request` is clearer...

Reply:
I renamed it to `KittyKeyboard` to match other commands such as `WindowSize`. Similar to `WindowSize` command, `KittyKeyboard` queries the terminal and returns a `KittyKeyboardMsg` in `Update`.